beginner_source/onnx/onnx_registry_tutorial.py

```diff
@@ -0,0 +1,469 @@
+#
+
+
+import torch
+print(torch.__version__)
+torch.manual_seed(191009)  # set the seed for reproducibility
+
+import onnxscript  # pip install onnxscript
+print(onnxscript.__version__)
+
+# NOTE: opset18 is the only version of ONNX operators we are
+# using in torch.onnx.dynamo_export for now.
+from onnxscript import opset18
+
+import onnxruntime  # pip install onnxruntime
+print(onnxruntime.__version__)
+
+
+######################################################################
+# Unsupported ATen operators
+# ---------------------------------
+#
+# ATen operators are implemented by PyTorch, and the ONNX exporter team must manually implement the
+# conversion from ATen operators to ONNX operators through [ONNX Script](https://onnxscript.ai/). Although the ONNX exporter
+# team has been making their best efforts to support as many ATen operators as possible, some ATen
+# operators are still not supported. In this section, we will demonstrate how you can implement any
+# unsupported ATen operators, which can contribute back to the project through the PyTorch GitHub repo.
+#
+# If the model cannot be exported to ONNX, for instance, :class:`aten::add.Tensor` is not supported
+# by ONNX The error message can be found, and is as follows (for example,  ``aten::add.Tensor``):
+#    ``RuntimeErrorWithDiagnostic: Unsupported FX nodes: {'call_function': ['aten.add.Tensor']}. ``
+#
+# To support unsupported ATen operators, we need two things:
+# * The unsupported ATen operator namespace, operator name, and the
+#    corresponding overload (for example ``<namespace>::<op_name>.<overload> - aten::add.Tensor``),
+#    which can be found in the error message.
+# * The implementation of the operator in `ONNX Script <https://github.com/microsoft/onnxscript>`__.
+#
+
+
+# NOTE: `is_registered_op` is a method in ONNX registry that checks
+# whether the operator is supported by ONNX. If the operator is not
+# supported, it will return False. Otherwise, it will return True.
+onnx_registry = torch.onnx.OnnxRegistry()
+# aten::add.default and aten::add.Tensor are supported by ONNX
+print(f"aten::add.default is supported by ONNX registry: \
+      {onnx_registry.is_registered_op(namespace='aten', op_name='add', overload='default')}")
+# aten::add.Tensor is the one invoked by torch.ops.aten.add
+print(f"aten::add.Tensor is supported by ONNX registry: \
+      {onnx_registry.is_registered_op(namespace='aten', op_name='add', overload='Tensor')}")
+
+
+######################################################################
+# In this example, we will assume that ``aten::add.Tensor`` is not supported by the ONNX registry,
+# and we will demonstrate how to support it. The ONNX registry allows user overrides for operator
+# registration. In this case, we will override the registration of ``aten::add.Tensor`` with our
+# implementation and verify it. However, this unsupported operator should return False when
+# checked with :meth:`onnx_registry.is_registered_op`.
+#
```

Comment:
```suggestion
# The ONNX registry allows user overrides for operator
# registration. In this case, we will override the registration of ``aten::add.Tensor`` with our
# implementation and verify it. However, this unsupported operator should return False when
# checked with :meth:`onnx_registry.is_registered_op`.
#
```

Moved part of the text to the beginning of the section

Reply:
As replied above.